app/services/food_service.rb

```diff
@@ -0,0 +1,23 @@
+class FoodService
+  def results(q)
```

Comment:
This method and the total_results method would be better off as class methods, so that Ruby has less garbage collection to do. https://scoutapm.com/blog/ruby-garbage-collection